src/ECharts/Legend.purs

```diff
@@ -0,0 +1,91 @@
+module ECharts.Legend where
+
+import Data.Maybe
+import Data.StrMap 
+import Data.Argonaut.Core
+import Data.Argonaut.Encode
+import Data.Argonaut.Combinators
+
+import ECharts.Color
+import ECharts.Common
+import ECharts.Coords
+import ECharts.Style.Text
+import ECharts.Formatter
+
+data LegendItem = LegendString String |
+  LegendItem {
+    "name" :: Maybe String,
+    "icon" :: Maybe String,
+    "textStyle" :: Maybe TextStyle
+    }
+
+instance legendItemEncodeJson :: EncodeJson LegendItem where
+  encodeJson (LegendString name) = fromString name
+  encodeJson (LegendItem obj) =
+    fromObject $ fromList $
+    [
+      "name" := obj.name,
+      "icon" := obj.icon,
+      "textStyle" := obj.textStyle
+    ]
+
+newtype Legend =
+  Legend {
+    "show" :: Maybe Boolean,
+    "orient" :: Maybe Orient,
+    "x" :: Maybe XPos,
+    "y" :: Maybe YPos,
+    "backgroundColor" :: Maybe Color,
+    "borderColor" :: Maybe Color,
+    "borderWidth" :: Maybe Number,
+    "padding" :: Maybe (Corner Number),
+    "itemGap" :: Maybe Number,
+    "itemHeight" :: Maybe Number,
+    "itemWidth" :: Maybe Number,
+    "textStyle" :: Maybe TextStyle,
+    "formatter" :: Maybe Formatter,
+    "selectedMode" :: Maybe SelectedMode,
+    "selected" :: Maybe (StrMap Boolean),
+    "data" :: Maybe [LegendItem]
```

Comment:
For types which have a natural zero, like array, if the library interprets `null` the same as `[]`, then it may make sense to omit the `Maybe`, to simplify construction. Otherwise there are two redundant zeros: `Nothing` and `Just []`.